vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -901,6 +901,7 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
           if ((typeof mode === 'string' || mode instanceof String) && validSelectionModes.indexOf(mode) >= 0) {
             selectionMode = mode;
             selectedKeys = {};
+            grid.selectedItems = [];
```

Comment:
Both `AbstractGridMultiSelectionModel` and `AbstractGridSingleSelectionModel` run a `deselectAll` when they are removed from the grid:
https://github.com/vaadin/flow-components/blob/0c2a684c5323276e384c14ce31f85f46c13c55d6/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java#L97-L104
https://github.com/vaadin/flow-components/blob/0c2a684c5323276e384c14ce31f85f46c13c55d6/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridSingleSelectionModel.java#L174-L178

That also removes client-side selection when switching between those two. However, when switching to the none selection mode, the connector prevents the selection from being modified: https://github.com/vaadin/flow-components/blob/0c2a684c5323276e384c14ce31f85f46c13c55d6/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts#L140-L143

Rather than add more branches to that method I just reset the selected items when changing the mode. Also decided to keep the `deselectAll` calls, technically the selection model instances are exposed with having only protected visibility and someone might rely on that behavior.